src/sync/mux.zig

```diff
@@ -251,6 +251,39 @@ pub fn RwMux(comptime T: type) type {
                 .valid = true,
             };
         }
+
+        pub fn readWithLock(self: *Self) struct { *const T, RLockGuard } {
+            var lock_guard = self.read();
+            const t = lock_guard.get();
+            return .{ t, lock_guard };
+        }
+
+        pub fn writeWithLock(self: *Self) struct { *T, WLockGuard } {
+            var lock_guard = self.write();
+            const t = lock_guard.mut();
+            return .{ t, lock_guard };
+        }
```

Comment:
i prefer the methods i used in this pr - its less code and more in line with other code you see (one method to lock + get and one method to unlock) - you also lose zls types when you do `item.get()`

i kept both approaches here because gossip uses the .get way and didnt want to touch it in this pr - but i think we should transition gossip to use this style too and probs change the `get` to a private method (.. im starting to warm up to private methods lol)